terraform/modules/lb/backends/lb_backends.tf

```diff
@@ -27,11 +27,41 @@ resource "oci_load_balancer_backend_set" "wls_lb_backendset" {
   lb_cookie_session_persistence_configuration {}
 }
 
+resource "oci_load_balancer_backend_set" "wls_lb_backendset_secure_mode" {
+  # If using existing load balancer in secured production mode, use per-created backend set of existing lb
+  count = var.use_existing_lb ? 0 : var.configure_secure_mode ? 1 : 0
+
+  name             = var.lb_backendset_name
+  load_balancer_id = var.load_balancer_id
+  policy           = var.lb_policy
+
+  health_checker {
+    port                = var.backend_port
+    protocol            = var.lb_protocol
+    response_body_regex = ".*"
+    url_path            = local.health_check_url_path
+    return_code         = var.return_code
+  }
+
+  ssl_configuration {
+    trusted_certificate_authority_ids = [var.root_ca_id]
+    verify_depth = 1
+    verify_peer_certificate = true
+  }
+
+  # Set the session persistence to lb-session-persistence with all default values.
+  lb_cookie_session_persistence_configuration {}
+
+  lifecycle {
+    ignore_changes = [ssl_configuration]
+  }
+}
+
 resource "oci_load_balancer_listener" "wls_lb_listener_https" {
   count                    = local.use_https_listener_count
   load_balancer_id         = var.load_balancer_id
   name                     = "${var.resource_name_prefix}_https"
-  default_backend_set_name = var.use_existing_lb ? var.lb_backendset_name : oci_load_balancer_backend_set.wls_lb_backendset[count.index].name
+  default_backend_set_name = var.use_existing_lb ? var.lb_backendset_name : var.configure_secure_mode ? oci_load_balancer_backend_set.wls_lb_backendset_secure_mode[0].name : oci_load_balancer_backend_set.wls_lb_backendset[count.index].name
```

Comment:
Why not use count.index for secure backendset?

Reply:
Done